SpatialGDK/Source/SpatialGDK/Private/Utils/LockingStatics.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/LockingStatics.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "LoadBalancing/AbstractLockingPolicy.h"
+#include "SpatialConstants.h"
+#include "Utils/SpatialStatics.h"
+
+#include "Containers/UnrealString.h"
+
+DEFINE_LOG_CATEGORY(LogLocking);
+
+namespace
+{
+bool CanProcessActor(const AActor* Actor)
+{
+	if (Actor == nullptr)
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on nullptr Actor is invalid."));
+		return false;
+	}
+
+	const UNetDriver* NetDriver = Actor->GetWorld()->GetNetDriver();
+	if (!NetDriver->IsServer())
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on a client is invalid. Actor: %s"), *GetNameSafe(Actor));
+		return false;
+	}
+
+	if (!Actor->HasAuthority())
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on a non-auth Actor is invalid. Actor: %s."),
+			*GetNameSafe(Actor));
+		return false;
+	}
+
+	return true;
+}
+} // anonymous namespace
+
+FLockingToken ULockingStatics::AcquireLock(AActor* Actor)
+{
+	if (!CanProcessActor(Actor) || !USpatialStatics::IsSpatialMultiWorkerEnabled(Actor->GetWorld()))
+	{
+		return FLockingToken{ SpatialConstants::INVALID_ACTOR_LOCK_TOKEN };
+	}
+
+	UAbstractLockingPolicy* LockingPolicy = Cast<USpatialNetDriver>(Actor->GetWorld()->GetNetDriver())->LockingPolicy;
```

Comment:
I assume that if we're using spatial world settings, we're guaranteed that our net driver is a spatial net driver?

Reply:
It doesn't error out if not in multi worker so presumably not.

Reply:
I'm more worried about running a spatial game without spatial networking enabled. Do we still get a spatial net driver then? Nothing in CanProcessActor assumes spatial, and the only relevant check in IsSpatialMultiWorkerEnabled is if we have spatial world settings.

Reply:
It's quite roundabout but the `SpatialStatics::IsSpatialMultiWorkerEnabled` will ultimately call `WorldSettings::IsMultiWorkerEnabled()` which checks whether Spatial networking is enabled. This should mean we ultimately don't need to have a conditional against the SpatialNetDriver

Reply:
Aha I overlooked that you added that check, makes sense.